src/app/clusters/ecosystem-information-server/ecosystem-information-server.h

```diff
@@ -212,6 +229,7 @@ class EcosystemInformationServer
     CHIP_ERROR EncodeDeviceDirectoryAttribute(EndpointId aEndpoint, AttributeValueEncoder & aEncoder);
     CHIP_ERROR EncodeLocationStructAttribute(EndpointId aEndpoint, AttributeValueEncoder & aEncoder);
 
+    MatterContext & mMatterContext;
```

Comment:
It seems a bit odd to have the class have a reference to itself. Is there a gain to having the server BE a context as well as having a reference to the context?

Reply:
I think the gain is that I always just call `mMatterContext.MarkDirty(...) without needing to have if conditions